src/Shared/StringCompatibilityExtensions.cs

```diff
@@ -8,5 +8,13 @@ internal static class StringCompatibilityExtensions
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static bool Contains(this string str, char search) =>
         str.Contains(search.ToString());
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static bool EndsWith(this string str, char search) =>
+        str?.Length switch
```

Comment:
Please remove the `?`, the .NET method doesn't allow null here since it's an instance method, the polyfill shouldn't have a different behavior.

(This can then be simplified into `str.Length > 0 && str[str.Length - 1] == search;`)